### PR TITLE
Gitlab fixes for V1.4

### PR DESCRIPTION
**Pull request by r-pai**
### Requirements

Gitlab Installer and Core Integration changes for v1.4. 

### Description of the Change

Changes done for V1.4 to work when Gitlab Installer changes is integrated.

### Benefits

The third scenario with V1.4 Gitlab changes is working end to end.
Gitlab as a new SCM.

### Possible Drawbacks

Code changes might effect other two scenarios .

### CODE REVIEW

File: delete-serverless-service-build-pack/Jenkinsfile [hunk 1]
```diff
@@ -38,20 +39,25 @@ node {
...
 	def var_bitbucket_branch_perm	= repo_protocol + var_bitbucket_base + "/rest/branch-permissions/2.0/projects/" + env.REPO_LOC + "/repos/"
 	def var_bitbucket_webhook_api	= repo_protocol + var_bitbucket_base + "/rest/webhook/1.0/projects/" + env.REPO_LOC + "/repos/"
 	def var_scm_source_hook 		= JenkinsLocationConfiguration.get().getUrl()+"/bitbucket-scmsource-hook/notify"
-	def jazzBuildModuleURL = "http://${var_bitbucket_base}/scm/${env.REPO_CORE}/jazz-build-module.git"
+	
+	//Setting SCM Repo
+	setSCMBase(scm, env.REPO_BASE)
+
+    def scm_base_url                = repo_protocol + scm_base + env.REPO_CORE + "/"
+    def var_cas_clone_url           = scm_base + env.REPO_LOC + "/"
```
Comment (suryajak) [inline delete-serverless-service-build-pack/Jenkinsfile]: variables naming consistency => cas_clone_url

File: delete-serverless-service-build-pack/Jenkinsfile [hunk 2]
```diff
@@ -38,20 +39,25 @@ node {
...
+	def configPackBBURL             = "http://" + scm_base + env.REPO_CORE + "/serverless-config-pack.git"
+    def var_serviceonboarding_repo 	= repo_protocol + scm_base + env.REPO_CORE + "/service-onboarding-build-pack.git"
```
Comment (suryajak) [inline delete-serverless-service-build-pack/Jenkinsfile]: variables naming consistency => serviceonboarding_repo

File: gitlab-build-pack/Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+ def jenkins_username  = env.JENKINS_USERNAME
+ def jenkins_password  = env.JENKINS_PASSWORD
+
+ def gitlab_repo_name  = env.gitlabSourceRepoName
+ def gitlab_branch     = env.gitlabBranch
+ def service_type = ''
+ def service_name = ''
+ def domain_name  = ''
+ 
+ echo "gitlab_credentialsId:gitlab_credentialsId"
```
Comment (suryajak) [inline gitlab-build-pack/Jenkinsfile]: Remove these echo statements

File: gitlab-build-pack/Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        //Add scm_project as cas
+        build_job = build_job + "&scm_project=cas"
+     } else if (service_type == 'lambda' || service_type == 'function'){
+        build_job = env.LAMBDA_BUILD_URI_DEV
+        if ( gitlab_branch == 'master') {
+            build_job = env.LAMBDA_BUILD_URI 
+        }
+    }
+
+    echo "service_type:$service_type"
```
Comment (suryajak) [inline gitlab-build-pack/Jenkinsfile]: remove debug statements

File: gitlab-build-pack/Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    echo "build_job:$build_job"
+    echo "domain_name:$domain_name"
+    echo "service_name:$service_name"
+                
+    def job_url = JenkinsLocationConfiguration.get().getUrl() + build_job
+    echo "job_url:$job_url"
```
Comment (suryajak) [inline gitlab-build-pack/Jenkinsfile]: remove

File: jenkins-build-pack-lambda/Jenkinsfile [hunk 6]
```diff
@@ -62,15 +63,19 @@ node ()  {
...
 	def bitbucket_name = params.service_name
 	def var_credentialsId 			= env.REPO_CREDENTIAL_ID
 
-	def scm_base_url = "http://${env.REPO_BASE}/scm/${env.REPO_LOC}/"
-	def var_api_key 				= env.API_KEY
+	//Setting SCM Repo
+	setSCMBase(scm, env.REPO_BASE)
+
+	def scm_base_url = "http://${scm_base}${env.REPO_LOC}/"
+	def var_api_key  = env.API_KEY
```
Comment (suryajak) [inline jenkins-build-pack-lambda/Jenkinsfile]: remove var prefix

File: service-onboarding-build-pack/Jenkinsfile [hunk 7]
```diff
@@ -60,7 +60,7 @@ node  {
 	def var_bitbucket_branch_perm	= repo_protocol + repo_base + "/rest/branch-permissions/2.0/projects/" + env.REPO_LOC + "/repos/"
 	def var_bitbucket_webhook_api	= repo_protocol + repo_base + "/rest/webhook/1.0/projects/" + env.REPO_LOC + "/repos/"
 	def jenkins_url                 = JenkinsLocationConfiguration.get().getUrl()
-	def bitbucket-scmscource-hook  	= jenkins_url + "/bitbucket-scmsource-hook/notify"
+	def bitbucket_scmscource_hook  	= jenkins_url + "/bitbucket-scmsource-hook/notify"
```
Comment (suryajak) [inline service-onboarding-build-pack/Jenkinsfile]: rename to scmsource_hook (we are no longer bitbucket specific)

File: service-onboarding-build-pack/Jenkinsfile [hunk 8]
```diff
@@ -460,7 +459,7 @@ def setRepoUrl(repo_protocol, scm_repo_name, repo_base)
 	return repo_url
 }
 
-def createProjectInSCM(gitlab_private_token, gitlab_cas_name_space_id, repo_base, scm_repo_name, username, password, cas_rest_repo)
+def createProjectInSCM(scm, gitlab_private_token, gitlab_cas_name_space_id, repo_base, scm_repo_name, username, password, cas_rest_repo)
```
Comment (suryajak) [inline service-onboarding-build-pack/Jenkinsfile]: The signature for this method looks wrong. Why would createProjectinSCM take a specific SCM's parameter (gitlab_private_token)?

Comment (suryajak) [reply service-onboarding-build-pack/Jenkinsfile]: I believe Daimen's story will fix this issue where we have scmmodule

File: gitlab-build-pack/Jenkinsfile [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+    	   //Find the service type	
+    	   if (fileExists('build.api')) {
+				service_type = "api"
+    	   } else if(fileExists('build.lambda')) {
+				service_type = "lambda"
+    	   } else if(fileExists('build.website')) {
+				service_type = "website"
+    	   }
+
+            //Read the JenkinsFile to find the service name and domain
```
Comment (suryajak) [inline gitlab-build-pack/Jenkinsfile]: This approach is not correct. We can't read a JenkinsFile and look for things to understand the service name and domain. 

Use deployment_env instead.

File: gitlab-build-pack/Jenkinsfile [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+    def job_url = JenkinsLocationConfiguration.get().getUrl() + build_job
+    echo "job_url:$job_url"
+ 
+    if ( service_type != '' && service_name != '' && domain_name != '') { 
+        echo "Call the build job"
+        sh "curl -X GET -k -v -u \"$jenkins_username:$jenkins_password\"  \"" + job_url + "&service_name=$service_name&domain=$domain_name&scm_branch=$gitlab_branch\""
+     }
+     else
+     {
+        echo "Error:ServiceType or ServiceName or DomainName is empty"
```
Comment (suryajak) [inline gitlab-build-pack/Jenkinsfile]: also call error on the job as this is a configuration issue somewhere and hence admins/someone should take a look.

File: delete-serverless-service-build-pack/Jenkinsfile (hunk 1 above)
Comment (r-pai) [reply delete-serverless-service-build-pack/Jenkinsfile]: Changed

File: delete-serverless-service-build-pack/Jenkinsfile (hunk 2 above)
Comment (r-pai) [reply delete-serverless-service-build-pack/Jenkinsfile]: Removed this as its not used in the code.

File: gitlab-build-pack/Jenkinsfile (hunk 3 above)
Comment (r-pai) [reply gitlab-build-pack/Jenkinsfile]: Removed

File: gitlab-build-pack/Jenkinsfile (hunk 4 above)
Comment (r-pai) [reply gitlab-build-pack/Jenkinsfile]: Removed

File: gitlab-build-pack/Jenkinsfile (hunk 5 above)
Comment (r-pai) [reply gitlab-build-pack/Jenkinsfile]: Removed

File: gitlab-build-pack/Jenkinsfile (hunk 10 above)
Comment (r-pai) [reply gitlab-build-pack/Jenkinsfile]: Done

File: jenkins-build-pack-lambda/Jenkinsfile (hunk 6 above)
Comment (r-pai) [reply jenkins-build-pack-lambda/Jenkinsfile]: Changed

File: service-onboarding-build-pack/Jenkinsfile (hunk 7 above)
Comment (r-pai) [reply service-onboarding-build-pack/Jenkinsfile]: Changed

File: gitlab-build-pack/Jenkinsfile (hunk 9 above)
Comment (r-pai) [reply gitlab-build-pack/Jenkinsfile]: Changed the approach by looking into deployment-dev.yml

**Pull request merged**